src/main/java/com/google/cloud/tools/appengine/api/deploy/DeployConfiguration.java

```diff
@@ -16,32 +16,68 @@
 
 package com.google.cloud.tools.appengine.api.deploy;
 
-import com.google.cloud.tools.appengine.api.Configuration;
+import com.google.auto.value.AutoValue;
 import java.io.File;
 import java.util.List;
 import javax.annotation.Nullable;
 
 /** Configuration for {@link AppEngineDeployment#deploy(DeployConfiguration)}. */
-public interface DeployConfiguration extends Configuration {
+@AutoValue
+public abstract class DeployConfiguration {
+  /** GCS storage bucket used for staging files associated with deployment. */
+  @Nullable
+  public abstract String getBucket();
 
+  /** List of deployable target directories or yaml files. */
   @Nullable
-  List<File> getDeployables();
+  public abstract List<File> getDeployables();
```

Comment:
Why are these suddenly public?

Reply:
It used to be an interface where everything was public by default.